### PR TITLE
ipykernel 6.6.1

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
Hmmm:

```
RuntimeError: Solver could not find solution.Mamba failed to solve:
 - jupyter_core <4.2
 - cross-python_osx-arm64
 - python 3.10.* *_cpython
 - jupyter_client <7.0
 - ipython >=5.0
 - debugpy >=1.0.0,<2.0

with channels:

The reported errors are:
- Encountered problems while solving:
-   - package debugpy-1.4.1-py310hba3363e_1 requires python_abi 3.10.* *_cp310, but none of the providers can be installed
- 

```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,21 +30,24 @@ requirements:
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - ipython >=5.0                          # [build_platform != target_platform]
     - jupyter_client <7.0                    # [build_platform != target_platform]
```
Comment (ccordoba12) [inline recipe/meta.yaml:32]:
```suggestion
    - jupyter_client <8.0                    # [build_platform != target_platform]
```

This constraint seems to be wrong. This  was probably added before jupyter_client 7 was stable, but now it shouldn't be necessary.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -31,21 +30,24 @@ requirements:
...
+    - jupyter_core <4.2                      # [build_platform != target_platform]
   {% endif %}
   host:
-    - debugpy >=1,<2.0
+    - debugpy >=1.0.0,<2.0
     - ipython >=7.23.1
     - jupyter_client <7.0
```
Comment (ccordoba12) [inline recipe/meta.yaml:38]:
```suggestion
    - jupyter_client <8.0
```

The same here.

**Review by ccordoba12 — COMMENTED** [review]
Maybe this is why mamba can't resolve dependencies correctly.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -30,22 +29,25 @@ requirements:
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
     - ipython >=5.0                          # [build_platform != target_platform]
-    - jupyter_client <7.0                    # [build_platform != target_platform]
+    - jupyter_client <8.0                    # [build_platform != target_platform]
+    - jupyter_core <4.2                      # [build_platform != target_platform]
```
[the remark this answers is not in the document]
Comment (Carreau) [reply recipe/meta.yaml:33]: 'cause we are currently at 4.9.1 and it's complaining about Python 2.7...

[the remark this answers is not in the document]
Comment (bollwyvl) [reply recipe/meta.yaml]: haha, well thanks!

**Comment by bollwyvl** [comment]
LGTM, any regrets?

**Comment by blink1073** [comment]
Thanks y'all!

**Pull request merged**